### PR TITLE
feat: eth_signTransaction and personal_sign components

### PR DESCRIPTION
**Pull request by JeneaVranceanu**
https://app.clickup.com/t/86byuf8j1

Added support for `eth_signTransaction` and `personal_sign`.
- `personal_sign` and `eth_sign` are now one component. The returned value must be the same as for `eth_sign`. 
- `eth_signTransaction` - visually it's a combination of `eth_sendTransaction` and `eth_sign`. This component sends a transaction to sign and allows for:
    - signer address recovery (same as `eth_sign` component);
    - signature validation through `EIP1271.isValidSignature` (same as `eth_sign` component);
    - checking if the signer has `SIGN` permission (just decided to add);

https://github.com/lukso-network/universalprofile-test-dapp/assets/36865532/fc99454f-5f11-49af-9b82-c50707121894

### CODE REVIEW

File: src/components/endpoints/SendTransaction.vue [hunk 1]
```diff
@@ -25,229 +20,6 @@ const data = ref<string>('')
 const hasData = ref(false)
 const isPending = ref(false)
 
-const methods: MethodSelect[] = [
```
Comment (JeneaVranceanu) [inline src/components/endpoints/SendTransaction.vue:28]: This was just moved to a separate file to be reused in `SignTransaction.vue` component.

**Review by richtera — APPROVED** [review]

**Comment by richtera** [comment]
Curious but should we upgrade the sign with ethereum stuff? I had looked at that before but it seemed to not match signatures somewhere

**Comment by JeneaVranceanu** [comment]
> it seemed to not match signatures somewhere

@richtera If so then we need to update it. I'll open a separate task to double-check that SIWE implementation matches the standard.

**Comment by JeneaVranceanu** [comment]
Sorry for the long waiting. Had more issues with tests.
Now it's fixed and ready to be re-reviewed and potentially merged.

File: src/views/Home.vue [hunk 2]
```diff
@@ -45,7 +46,9 @@ const hasExtension = computed(() => !!window.lukso)
     </div>
     <div class="tile is-ancestor">
       <Sign />
+      <PersonalSign />
       <GetNetworkId />
+      <!-- <SignTransaction /> -->
```
Comment (dzbo) [inline src/views/Home.vue:51]: Why commented out?

**Review by dzbo — COMMENTED** [review]
`Sign` and Personal Sign` are very similar, can we merge it into one? Just add another checkbox under `Sign in with Ethereum`.

**Comment by JeneaVranceanu** [comment]
> `Sign` and Personal Sign` are very similar, can we merge it into one? Just add another checkbox under `Sign in with Ethereum`.

I'll need to re-review this PR but yes, we can merge them into one.

**Comment by Hugoo** [comment]
I guess this is stale - shall we close this one? @JeneaVranceanu

**Comment by JeneaVranceanu** [comment]
> I guess this is stale - shall we close this one? @JeneaVranceanu

I'll check and let you know.

File: src/components/endpoints/SignTransaction.vue
Comment (JeneaVranceanu) [inline src/components/endpoints/SignTransaction.vue:1]: Keeping this component is optional as we do not support eth_signTransaction in the UP extension at the moment. https://app.clickup.com/t/2645698/DEV-8191

**Review by richtera — APPROVED** [review]
This looks ok, but it's only logging things?

**Comment by franckwei** [comment]
@JeneaVranceanu FYI this PR has been opened since last summer, should we merge?

**Comment by JeneaVranceanu** [comment]
> @JeneaVranceanu FYI this PR has been opened since last summer, should we merge?

I'm moving the task to testing. 
I think we will merge tomorrow.
It's kind of the lowest priority 😅

**Pull request merged**